internal/daemon/worker/worker.go

```diff
@@ -188,26 +188,81 @@ func (w *Worker) Start() error {
 		return fmt.Errorf("error starting worker listeners: %w", err)
 	}
 
-	nodeCreds, err := types.NewNodeCredentials(w.baseContext, w.WorkerAuthStorage, nodeenrollment.WithWrapper(w.conf.WorkerAuthStorageKms))
-	if err != nil {
-		return fmt.Errorf("error generating new node creds: %w", err)
-	}
+	// Check for worker auth creds and if not create new ones
+	var createNodeAuthCreds bool
+	var createFetchRequest bool
+	nodeCreds, err := types.LoadNodeCredentials(w.baseContext, w.WorkerAuthStorage, nodeenrollment.CurrentId, nodeenrollment.WithWrapper(w.conf.WorkerAuthStorageKms))
+	switch {
+	case err == nil:
+		if nodeCreds == nil {
+			event.WriteSysEvent(w.baseContext, op, "no error loading worker auth creds but nil creds, creating new creds for registration")
```

Comment:
are you seeing this event?  just curious if you manually tested it -> no test coverage :-)  BTW, the test coverage is great overall for these new bits.  so hard to test failure modes.

Reply:
I'm not, no. I don't even know how it'd happen given that it should never return nil and a non-nil error so this is just to prevent panic in some absolutely bonkers case.